client/state/posts/test/actions.js

```diff
@@ -391,25 +392,29 @@ describe( 'actions', () => {
 		} );
 
 		test( 'should dispatch failure action when error occurs while saving new post', () => {
-			return savePost( 77203074, null, { title: 'Hello World' } )( spy ).then( () => {
-				expect( spy ).to.have.been.calledWith( {
-					type: POST_SAVE_FAILURE,
-					siteId: 77203074,
-					postId: null,
-					error: sinon.match( { message: 'User cannot edit posts' } ),
+			return savePost( 77203074, null, { title: 'Hello World' } )( spy )
+				.catch( noop )
```

Comment:
Why the `.catch( noop )` here and elsewhere?  If we hit the `catch` block, shouldn't the test actually fail?

Reply:
See the description of #23359 for the explanation. The `savePost` dispatch fires the appropriate Redux actions on success and error, and returns the resolved/rejected promise so that the component can update its local state accordingly, too.

What we're testing there is: after the dispatch is finished, no matter if it was success or error, check if the expected actions were dispatched.

The best solution would be `savePost()( spy ).finally( () => { expect( ... ) } )`, but `finally` is very new and is not available everywhere. What is most important, it's not in the version of Node.js we use for running the tests.